tests/test_connection.py

```diff
@@ -434,6 +434,43 @@ def test_mfa_passcode(mock_connect, runner, command):
     assert kwargs["passcode"] == "123"
 
 
+def test_if_password_callback_is_called_only_once_from_prompt(runner):
+    with NamedTemporaryFile("w+", suffix=".toml") as tmp_file:
+        result = runner.invoke_with_config_file(
+            tmp_file.name,
+            [
+                "connection",
+                "add",
+            ],
+            input="connName\naccName\nuserName\npassword",
+        )
+
+    assert result.exit_code == 0
+    assert result.output.count("WARNING!") == 1
```

Comment:
Should we log warning if password is passed via prompt?

Reply:
You mean - update callback function, to log it?

Reply:
This test case is for password provided from prompt. I'm just wondering if we should log a warning in such case? The warning message says only about insecure `--password` flag, it doesn't say anything about prompt: `WARNING! Using --password via the CLI is insecure. Use environment variables instead.`

Reply:
Well, in my understanding, we discourage the users from using open-text passwords. Like in the prompt or flag, because they will be stored in an openly avaiable `config.toml`. So in both cases we should warn users, but maybe i should change warning text to cover both cases

Reply:
My understanding was different. IMO the goal of this warning was to discourage the users from using plain-text password included as part of CLI command (so from using `--password` option), because executed CLI commands can be visible in operating system's logs.
IMO we can't discourage users from using plain-text passwords in `config.toml` because we don't give them a good alternative.

@sfc-gh-turbaszek Can you share your thought on this?

Reply:
Using `--password=XYZ` should cause a warning. Passing password via config or prompt is relatively safe. So if `--password` opens a prompt that's ok